test/e2e/nodepool_machineconfig_test.go

```diff
@@ -163,13 +162,36 @@ func testNodepoolMachineconfigGetsRolledout(parentCtx context.Context, mgmtClien
 			t.Fatalf("failed to update nodepool %s after adding machineconfig: %v", nodePool.Name, err)
 		}
 
+		// DS Customization
 		ds := machineConfigUpdatedVerificationDS.DeepCopy()
+		dsName := ds.Name + "-replace"
```

Comment:
nit: refactor this into a function that returns a daemonset given a daemonset name and nodepool name and call that function from here and anywhere else you need to run the daemonset.

Reply:
changed, PTAL when you have some time :)